libraries/AP_HAL_ChibiOS/hwdef/StampH743/hwdef.dat

```diff
@@ -0,0 +1,237 @@
+PH5 IMU2_CS CS
+PA10 IMU2_DRDY INPUT
+
+# SPI3 - IMU3 ICM-42688-P
+PB2  SPI3_MOSI SPI3
+PC10 SPI3_SCK  SPI3
+PC11 SPI3_MISO SPI3
+PI4 IMU3_CS CS
+PI6 IMU3_DRDY INPUT
+
+# SPI4 - External
+PE2 SPI4_SCK SPI4
+PE5 SPI4_MISO SPI4
+PE6 SPI4_MOSI SPI4
+PI7 SP4_DRDY INPUT
+PE4 SP4_CS1 CS
+
+# PWM output pins
+PH10 TIM5_CH1 TIM5 PWM(1) GPIO(50) BIDIR
+PH11 TIM5_CH2 TIM5 PWM(2) GPIO(51)
+PH12 TIM5_CH3 TIM5 PWM(3) GPIO(52) BIDIR
+PI0  TIM5_CH4 TIM5 PWM(4) GPIO(53)
+PA15 TIM2_CH1 TIM2 PWM(5) GPIO(54) BIDIR
+PB3  TIM2_CH2 TIM2 PWM(6) GPIO(55)
+PB10 TIM2_CH3 TIM2 PWM(7) GPIO(56) BIDIR
+PD12 TIM4_CH1 TIM4 PWM(8) GPIO(57) BIDIR
+PD13 TIM4_CH2 TIM4 PWM(9) GPIO(58)
+PD14 TIM4_CH3 TIM4 PWM(10) GPIO(59)
+PD15 TIM4_CH4 TIM4 PWM(11) GPIO(60)
+# we need to disable DMA on the last FMU channel
+# as timer 14 doesn't have a TIMn_UP DMA option
+PF9 TIM14_CH1 TIM14 PWM(12) GPIO(61) NODMA
+
+# CAN bus
+PD0  CAN1_RX CAN1
+PD1  CAN1_TX CAN1
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+# I2C buses
+
+# I2C1 - BMM150
+PB9 I2C1_SDA I2C1
+PB8 I2C1_SCL I2C1
+
+# I2C2 - DPS310
+PF1 I2C2_SCL I2C2
+PF0 I2C2_SDA I2C2
+
+# I2C3 - GPS1 external compass
+PA8 I2C3_SCL I2C3
+PH8 I2C3_SDA I2C3
+
+# I2C4 - GPS2 external compass
+PF14 I2C4_SCL I2C4
+PF15 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C1 I2C2 I2C3 I2C4
+define HAL_I2C_INTERNAL_MASK 3
+
+# power enable pins
+#PI11 IMU1_PWR_EN OUTPUT HIGH
+#PF4 IMU2_PWR_EN OUTPUT HIGH
+#PE7 IMU3_PWR_EN OUTPUT HIGH
+#PG8 MAG_PWR_EN OUTPUT HIGH
+#PC13 SD_PWR_EN OUTPUT HIGH
+#PG15 Ethernet_PWR_EN OUTPUT HIGH
+
+# microSD support
+PD6  SDMMC2_CK SDMMC2
+PD7  SDMMC2_CMD SDMMC2
+PB14 SDMMC2_D0 SDMMC2
+PB15 SDMMC2_D1 SDMMC2
+PG11 SDMMC2_D2 SDMMC2
+PB4  SDMMC2_D3 SDMMC2
+define FATFS_HAL_DEVICE SDCD2
+
+# safety
+PF5 SAFETY_IN INPUT PULLDOWN
+
+# LED
+PE3 LED_RED OUTPUT OPENDRAIN GPIO(90) HIGH
+
+define HAL_GPIO_A_LED_PIN 90
+define HAL_GPIO_LED_ON 0
+
+# barometers
+BARO DPS310 I2C:1:0x76
+
+# compass
+COMPASS BMM150 I2C:0:0x10 false ROTATION_NONE
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+
+# IMUs
+SPIDEV icm42670   SPI1 DEVID1  IMU1_CS MODE3  2*MHZ  8*MHZ
+SPIDEV icm42688   SPI2 DEVID1  IMU2_CS MODE3  2*MHZ  16*MHZ
+SPIDEV icm42688_2 SPI3 DEVID1  IMU3_CS MODE3  2*MHZ  16*MHZ
+
+IMU Invensensev3 SPI:icm42688 ROTATION_PITCH_180_YAW_90
+IMU Invensensev3 SPI:icm42688_2 ROTATION_PITCH_180_YAW_90
+IMU Invensensev3 SPI:icm42670 ROTATION_YAW_90
+
+define HAL_DEFAULT_INS_FAST_SAMPLE 7
+
+# use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+define HAL_STORAGE_SIZE 32768
+STORAGE_FLASH_PAGE 14
+
+DMA_PRIORITY SDMMC* TIM* SPI1*
+DMA_NOSHARE TIM2* TIM5*
+
+# enable FAT filesystem support (needs a microSD defined via SDMMC)
+define HAL_OS_FATFS_IO 1
+
+# enable DFU reboot for installing bootloader
+# note that if firmware is build with --secure-bl then DFU is
+# disabled
+ENABLE_DFU_BOOT 1
+
+# build ABIN for flash-from-bootloader support:
+env BUILD_ABIN True
+
+# ethernet consumes a big chunk of flash
+include ../include/minimize_common.inc
```

Comment:
likely better to change env OPTIMIZE to -Os

Reply:
Fixed